packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -845,6 +845,57 @@ describe('TokensController', () => {
       );
     });
 
+    it('should add tokens to allIgnoredTokens state only if we are not using current network', async () => {
```

Comment:
Is this test name as accurate as it could be?

- It says that tokens should only be added to `allIgnoredTokens` state, but we are never checking that state. We are just checking that `tokens` and `ignoredTokens` are not affected, which is a slightly different thing to check.
- Who are "we" in this context?

Perhaps a better name would be:

``` suggestion
    it('should not update detectedTokens, tokens, and ignoredTokens state for a network that is different from the globally selected network', async () => {
```

Reply:
On top of this, do we have a test which ensures that `detectedTokens`, `tokens`, and `ignoredTokens` _do_ get updated when passed the globally selected network? I'm not quite seeing that.

Reply:
suggestions implemented with two tests scenarios [7d0cf7d](https://github.com/MetaMask/core/pull/5014/commits/7d0cf7dcbaa5f8a37f7918fc9d54548053a0fc0e)
- should not update detectedTokens, tokens, and ignoredTokens state given a network that is different from the globally selected network
- should update tokens, and ignoredTokens and detectedTokens state for the globally selected network

Reply:
for detected tokens, it's a bit complex to test them, so I've put some assertions to make sure they're not affected, but I'm open to suggestions.